doc/manual/modules.rst

```diff
@@ -236,7 +236,29 @@ Namespace miscellanea
 ---------------------
 
 If a name is qualified (e.g. ``Base.sin``), then it can be accessed even if
-it is not exported. This is often useful when debugging.
+it is not exported. This is often useful when debugging. Normally Julia
+variables, functions, types, and the fields of composite types are publicly
+accessible. Julia does not have corresponding "private" versions of these
+items that are inaccessible from outside the module. However, a common
+Julia convention is that names prefixed with an underscore (e.g.,
```

Comment:
I'm not at all wedded to this language.  Perhaps just "However, names prefixed with an underscore are private implementation details." 

However, I just grepped base for " _" and found ~600 lines that look like names (primarily functions) intended to be private.  For example, osutils.jl:12 has a function `_os_test()` that looks like it's used only in the macros just below, and it seems like it's intended only for internal consumption. So it does seem to have crept in there.